apps/spruce/src/pages/projectSettings/tabs/GithubAppSettingsTab/getFormSchema.tsx

```diff
@@ -12,8 +42,106 @@ export const getFormSchema = (): ReturnType<GetFormSchema> => ({
       tokenPermissionRestrictions: {
         type: "object" as "object",
         title: "Token Permission Restrictions",
+        properties: {
+          permissionsByRequester: {
+            type: "array" as "array",
+            items: {
+              type: "object" as "object",
+              properties: {
+                requesterType: {
+                  type: "string" as "string",
+                  title: "",
+                },
+                permissionGroup: {
+                  type: "string" as "string",
+                  title: "",
+                  default: "",
```

Comment:
[nit] This clarifies the intention a bit, I think
```suggestion
                  default: allPermissionsGroup,
```